src/test/java/seedu/address/logic/commands/project/DeleteTaskCommandTest.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+
+    @Test
+    public void execute_invalidIndexFilteredTaskList_throwsCommandException() {
+        model.enter(project);
+
+        project.updateTaskFilter(task -> true);
+        Task taskToDelete = project.getFilteredSortedTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        // Update the task filter inside the project so that the taskToDelete is the only task shown
+        project.updateTaskFilter(task -> task.getTaskName().contains(taskToDelete.getTaskName()));
+        Index outOfBoundIndex = INDEX_SECOND_TASK;
+        // ensures that outOfBoundIndex is still in bounds of the whole task list
+        assertTrue(outOfBoundIndex.getZeroBased() < project.getTasks().size());
+
+        DeleteTaskCommand deleteTaskCommand = new DeleteTaskCommand(outOfBoundIndex);
+
+        assertCommandFailure(deleteTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredTaskList_success() {
+        model.enter(project);
+
+        project.updateTaskFilter(task -> true);
+        Task taskToDelete = project.getFilteredSortedTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        // Update the task filter inside the project so that the taskToDelete is the only task shown
+        project.updateTaskFilter(task -> task.getTaskName().contains(taskToDelete.getTaskName()));
+        DeleteTaskCommand deleteTaskCommand = new DeleteTaskCommand(INDEX_FIRST_TASK);
+
+        // Assign the taskToDelete to Alice so that
+        // we can check whether DeleteTaskCommand will remove the task in the participation
+        aliceParticipation.addTask(taskToDelete);
+        taskToDelete.addAssignee(ALICE.getGitUserNameString());
+
+        String expectedMessage = String.format(DeleteTaskCommand.MESSAGE_DELETE_TASK_SUCCESS, taskToDelete);
+
+        Model expectedModel = new ModelManager(model.getProjectCatalogue(), new UserPrefs());
+        Project expectedProject = ProjectBuilder.makeCopy(project);
+        expectedModel.setProject(project, expectedProject);
+        expectedProject.deleteTask(taskToDelete);
+
+        assertCommandSuccess(deleteTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+
+    @Test
+    public void equals() {
+        DeleteTaskCommand deleteFirstCommand = new DeleteTaskCommand(INDEX_FIRST_TASK);
+        DeleteTaskCommand deleteSecondCommand = new DeleteTaskCommand(INDEX_SECOND_TASK);
+
+        // same object -> returns true
+        assertTrue(deleteFirstCommand.equals(deleteFirstCommand));
+
+        // same values -> returns true
+        DeleteTaskCommand deleteFirstCommandCopy = new DeleteTaskCommand(INDEX_FIRST_TASK);
+        assertTrue(deleteFirstCommand.equals(deleteFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(deleteFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(deleteFirstCommand.equals(null));
+
+        // different task -> returns false
```

Comment:
What is the purpose of this test if I may ask?

Reply:
Test the equal method of delete command

Reply:
But what's the point, I mean
If it is equal or not, does it do anything useful?

Reply:
? So that it will not go wrong when comparing two deletetask command. This is also for higher coverage. U can also check delete project command Test adapted from the original ab3 one.